asmcomp/cmmgen.ml

```diff
@@ -60,6 +65,15 @@ let add_unboxed_id id unboxed_id bn env =
     unboxed_ids = V.add id (unboxed_id, bn) env.unboxed_ids;
   }
 
+let add_catch_types i typs env =
+  { env with
+    catch_types = IntMap.add i typs env.catch_types
+  }
+
+let catch_types i env =
+  try IntMap.find i env.catch_types
+  with Not_found -> []
```

Comment:
That should probably be a fatal_error here, rather than a probable one in the List.map2 bellow

Reply:
There are a number of `ccatch` created elsewhere (e.g. for `Ufor`), and they all have an empty list of arguments.

Reply:
Anyway, this will be for another PR!

Reply:
You don't have clambda version for those catches, so catch_types never get called for them.